### PR TITLE
Fix deprecated rsvg calls in newer versions of the lib.

### PR DESCRIPTION
**Pull request by nieknooijens**
ubuntu 22.04 comes with rsvg 2.52

in 2.46 rsvg_handle_get_dimensions() was deprecated and superseded by rsvg_handle_get_intrinsic_size_in_pixels

in 2.52 rsvg_handle_render_cairo() has been deprecated and superseded by rsvg_handle_render_document()

here's a quick fix.

### What does this PR do?

check whether a newer version of RSVG is installed and use the new calls instead.

### Closes Issue(s)
no issue yet, since it's a quick fix

### Motivation
keeping performous working on newer rsvg versions which will remove those calls in the future

### Additional Notes
new rsvg_handle_render_document() needs a viewport in which to render. we can all agree this should start at coordinates 0,0 but im not sure about the width/height, I took the ones coming from rsvg_handle_get_intrinsic_size_in_pixels() and apparently I guessed correctly since I don't see any visible glitches, would still be nice if someone can confirm this guess is correct.

### CODE REVIEW

**Comment by Lord-Kamina** [comment]
According to the documentation, you are correct. https://developer-old.gnome.org/rsvg/unstable/rsvg-Using-RSVG-with-cairo.html#rsvg-handle-render-cairo

**Comment by nieknooijens** [comment]
> According to the documentation, you are correct. https://developer-old.gnome.org/rsvg/unstable/rsvg-Using-RSVG-with-cairo.html#rsvg-handle-render-cairo

ok well then great! Guess I got good at guessing then. Or I start to understand the minds of software developers....

**Comment by Lord-Kamina** [comment]
I think you botched the actual version numbers in the preprocessor checks though.

**Review by Lord-Kamina — APPROVED** [review]
LGTM

File: game/svg.cc [hunk 1]
```diff
@@ -27,11 +27,17 @@ void loadSVG(Bitmap& bitmap, fs::path const& filename) {
 		g_error_free(pError);
 		throw std::runtime_error("Unable to load " + filename.string());
 	}
+#if LIBRSVG_MAJOR_VERSION >= 2 && LIBRSVG_MINOR_VERSION >= 52
+	gdouble svg_width = 0, svg_height = 0;
+	rsvg_handle_get_intrinsic_size_in_pixels(svgHandle.get(),&svg_width, &svg_height);
+	bitmap.resize(static_cast<int>(svg_width*factor + 0.5), static_cast<int>(svg_height *factor + 0.5));
```
Comment (Baklap4) [inline game/svg.cc:33]: missing another space between the multiplications
```suggestion
	bitmap.resize(static_cast<int>(svg_width * factor + 0.5), static_cast<int>(svg_height * factor + 0.5));
```

File: game/svg.cc [hunk 2]
```diff
@@ -27,11 +27,17 @@ void loadSVG(Bitmap& bitmap, fs::path const& filename) {
 		g_error_free(pError);
 		throw std::runtime_error("Unable to load " + filename.string());
 	}
+#if LIBRSVG_MAJOR_VERSION >= 2 && LIBRSVG_MINOR_VERSION >= 52
+	gdouble svg_width = 0, svg_height = 0;
+	rsvg_handle_get_intrinsic_size_in_pixels(svgHandle.get(),&svg_width, &svg_height);
```
Comment (Baklap4) [inline game/svg.cc:32]: Missing space
```suggestion
	rsvg_handle_get_intrinsic_size_in_pixels(svgHandle.get(), &svg_width, &svg_height);
```

File: game/svg.cc [hunk 3]
```diff
@@ -40,7 +46,12 @@ void loadSVG(Bitmap& bitmap, fs::path const& filename) {
 	  cairo_surface_destroy);
 	std::shared_ptr<cairo_t> dc(cairo_create(surface.get()), cairo_destroy);
 	cairo_scale(dc.get(), factor, factor);
+#if LIBRSVG_MAJOR_VERSION >= 2 && LIBRSVG_MINOR_VERSION >= 52
+	RsvgRectangle viewport = {0,0, svg_width, svg_height}; //viewport start is defenetely 0,0 but width and height?... from rsvg_handle_get_intrinsic_size_in_pixels I guess...
```
Comment (Baklap4) [inline game/svg.cc:50]:
```suggestion
	RsvgRectangle viewport = {0,0, svg_width, svg_height}; //viewport start is definitely 0,0 but width and height?... from rsvg_handle_get_intrinsic_size_in_pixels I guess...
```

**Review by Baklap4 — CHANGES_REQUESTED** [review]
A couple of minor things regarding code style

File: game/svg.cc (hunk 3 above)
Comment (nieknooijens) [reply game/svg.cc:50]: that comment can be removed alltogether tho

**Review by Baklap4 — APPROVED** [review]
LGTM

**Pull request merged**